primitives/io/src/lib.rs

```diff
@@ -279,7 +279,33 @@ pub trait DefaultChildStorage {
 		storage_key: &[u8],
 	) {
 		let child_info = ChildInfo::new_default(storage_key);
-		self.kill_child_storage(&child_info);
+		self.kill_child_storage(&child_info, None);
+	}
+
+	/// Clear at most `limit` keys from the specified child storage.
+	///
+	/// Deletes all keys from the overlay and up to `limit` keys from the backend.
```

Comment:
This is tricky: it will drop all child trie current changes without applying limit, and then build a removal payload of only `limit` key.
It make sense for the use case, but complicates the api.
But I don't really have a better idea. We could apply 'limit' to existing changes but that won't really make a lot more sense and would add to the code complexity for little good.

Maybe making it explicit that backend is state at start of block processing, I really don't know here.

Reply:
This behaviour serves the intended use case and is also stated explicitly in the docs.